content/plugins/current/meta/portmap.md

```diff
@@ -116,27 +130,94 @@ chain performs that masquerading. By default, bit 13 is set, but this is
 configurable. If you are using other tools that also use the iptables mark,
 you should make sure this doesn't conflict.
 
-Some container runtimes, most notably Kubernetes, already have a set of rules
-for masquerading when a specific mark bit is set. If so enabled, the plugin
-will use that chain instead.
-
 `POSTROUTING`:
 - `-j CNI-HOSTPORT-MASQ`
 
 `CNI-HOSTPORT-MASQ`:
 - `--mark 0x2000 -j MASQUERADE`
 
-Because MASQUERADE happens in POSTROUTING, it means that packets with source ip
-127.0.0.1 need to first pass a routing boundary before being masqueraded. By
-default, that is not allowed in Linux. So, the plugin needs to enable the sysctl
-`net.ipv4.conf.IFNAME.route_localnet`, where IFNAME is the name of the host-side
-interface that routes traffic to the container.
+## Rule structure (nftables)
+The organization is slightly simpler than in the iptables case. All
+rules are created in the `cni_hostport` table (of the `ip` or `ip6`
+family, as appropriate).
+
+### DNAT
+The DNAT rule rewrites the destination port and address of new connections.
+DNAT rules are added to the `hostports` or `hostip_hostports` chains
+of the `cni_hostport` table, depending on whether the mapping is for
+all host IPs or only for a single host IP.
+
+So, if a single container exists on IP 172.16.30.2/24 with ports 8080
+and 8043 on the host forwarded to ports 80 and 443 in the container,
+the rules look like this:
+
+```
+table ip cni_hostport {
+    comment "CNI portmap plugin"; }
+
+    chain input {
+        type nat hook input priority dstnat;
+        jump hostports
+    }
+    chain output {
+        type nat hook output priority dstnat;
+        fib daddr type local jump hostports
+    }
+
+    chain hostports {
+        ip protocol tcp th dport 8080  dnat ip addr . port to 172.16.30.2 . 80
+        ip protocol tcp th dport 8043  dnat ip addr . port to 172.16.30.2 . 443
+    }
+}
+
+New connections to the host will have to traverse every rule, so large numbers
+of port forwards may have a performance impact. This won't affect established
+connections, just the first packet.
+
+### SNAT (Masquerade)
+Some packets also need to have the source address rewritten:
+* connections from localhost
+* Hairpin traffic back to the container.
+* Plugins which traffic not go though default net namespace e.g., ipvlan,macvlan,etc. (need `masqAll` option)
```

Comment:
```suggestion
* Plugins which traffic not go through default net namespace e.g., ipvlan,macvlan,etc. (need `masqAll` option)
```

Reply:
(this was just copied from the existing iptables docs... I fixed it in both places)